presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -283,6 +284,11 @@ public SystemSessionProperties(
                         featuresConfig.isLegacyOrderBy(),
                         false),
                 booleanSessionProperty(
+                        SUPPORT_NULL_ON_LHS_OF_SEMI_JOIN,
```

Comment:
IIRC the code will push `y=3 or y is null` but it is correct *only if* we know `x` is not null. Which we don't currently know. Otherwise there wouldn't be any correctness issue nor this PR :)